pytorch_lightning/trainer/callback_hook.py

```diff
@@ -299,10 +306,16 @@ def on_load_checkpoint(self, checkpoint):
         # https://github.com/pytorch/xla/issues/2773
         if callback_states is not None:
             for callback in self.callbacks:
-                state = callback_states.get(type(callback))
-                if state:
-                    state = deepcopy(state)
```

Comment:
we shouldn't call the load checkpoint if the state is empty / not present, unless we also save empty states too. Maybe we should do that, I don't know. In any case, users have to touch all their implementations of `on_load_checkpoint`, which is not great :(

Reply:
We have other callback saving/loading related changes planned for 1.4 #6467.